### PR TITLE
Simplify CI Testing

### PR DESCRIPTION
**Pull request by Mpdreamz**
Ensures we can call the test suite grouping we partitioned on CI locally easily as well:

```bash

# the default is Unit e.g so the first two are equivalent
./build.sh test 
./build.sh test --test-suite Unit
./build.sh test --test-suite Integrations
./build.sh test --test-suite Profiler
./build.sh test --test-suite Azure
./build.sh test --test-suite IIS
./build.sh test --test-suite StartupHooks
```

`StartupHooks` and `Profiler` will build their dependencies automatically simplifying reproducing these further. You can use `--clean false` locally to run the tests multiple times without cleaning and rebuilding the artifacts (including profiler and agent zips).

This now also more clearly separates `Unit` from `Integrations`. Allowing us to more clearly run faster tests locally during our development loop for quick validations.

Lastly this moves us over to https://github.com/Tyrrrz/GitHubActionsTestLogger for github test reporting. This allows us to move away from stashing xml files and we can thus remove the `-reporter.yml` workflows.

### CODE REVIEW

**Review by gregkalapos — APPROVED** [review]

File: .github/workflows/test-linux.yml [hunk 1]
```diff
@@ -124,16 +116,10 @@ jobs:
...
-
     - name: 'Tests: StartupHooks'
-      uses: ./.github/workflows/test
-      with:
-        name: 'startuphooks'
-        project: 'test/startuphook/Elastic.Apm.StartupHook.Tests/Elastic.Apm.StartupHook.Tests.csproj'
-        
-
+      shell: bash
+      run: ./build.sh test --test-suite StartupHooks
```
Comment (v1v) [inline .github/workflows/test-linux.yml:121]: `StartupHooks` should be lower-cased instead? I see that pattern in the other places when using `--test-suite`

File: .github/workflows/test-linux.yml [hunk 2]
```diff
@@ -145,14 +131,9 @@ jobs:
...
-    - name: Build profiler
-      run: ./build.sh profiler-zip
-
     - name: 'Tests: Profiler'
-      uses: ./.github/workflows/test
-      with:
-          name: 'profiler'
-          project: 'test/profiler/Elastic.Apm.Profiler.Managed.Tests/Elastic.Apm.Profiler.Managed.Tests.csproj'
+      shell: bash
+      run: ./build.sh test --test-suite Profiler
```
Comment (v1v) [inline .github/workflows/test-linux.yml:136]: ditto

File: .github/workflows/test-windows.yml [hunk 3]
```diff
@@ -38,21 +38,18 @@ jobs:
...
-        uses: ./.github/workflows/test
-        with:
-          name: 'unit'
-          filter: 'FullyQualifiedName!~Elastic.Apm.StartupHook.Tests&FullyQualifiedName!~Elastic.Apm.Profiler.Managed.Tests&FullyQualifiedName!~Elastic.Apm.Azure'
-          framework: '' # reset to test all .NET frameworks including net462
+        run: ./build.bat test --test-suite Unit
+        
+      - name: 'Tests: Integrations'
+        shell: bash
+        run: ./build.bat test --test-suite Integrations
```
Comment (v1v) [inline .github/workflows/test-windows.yml:52]: Linux uses lowercase:

> run: ./build.sh test --test-suite integrations

File: .github/workflows/test-windows.yml [hunk 4]
```diff
@@ -62,14 +59,8 @@ jobs:
...
 
-      - name: Build agent-zip
-        run: ./build.bat agent-zip
-        
       - name: 'Tests: StartupHooks'
-        uses: ./.github/workflows/test
-        with:
-          name: 'startuphooks'
-          project: 'test/startuphook/Elastic.Apm.StartupHook.Tests/Elastic.Apm.StartupHook.Tests.csproj'
+        run: ./build.bat test --test-suite StartupHooks
```
Comment (v1v) [inline .github/workflows/test-windows.yml:63]: ditto

File: .github/workflows/test-windows.yml [hunk 5]
```diff
@@ -82,17 +73,11 @@ jobs:
...
         with:
           token: ${{ secrets.TC_CLOUD_TOKEN }}
 
       - name: 'Tests: Profiler'
-        uses: ./.github/workflows/test
-        with:
-          name: 'profiler'
-          project: 'test/profiler/Elastic.Apm.Profiler.Managed.Tests/Elastic.Apm.Profiler.Managed.Tests.csproj'
\ No newline at end of file
+        run: ./build.bat test --test-suite Profiler
```
Comment (v1v) [inline .github/workflows/test-windows.yml:83]: ditto :)

**Review by v1v — APPROVED** [review]
Minor comments, other than that LGTM

File: .github/workflows/test-windows.yml (hunk 5 above)
Comment (Mpdreamz) [reply .github/workflows/test-windows.yml]: Thanks @v1v  I don't know why this didn't trigger my code ocd 🙈

**Review by v1v — DISMISSED** [review]
LGTM.

qq: I see the `IIS` test-suite, but no changes in https://github.com/elastic/apm-agent-dotnet/blob/main/.github/workflows/test-windows-iis.yml , is that expected?

**Comment by Mpdreamz** [comment]
Good catch! I was planning to follow up this and #2363 merging IIS into `test-windows.yml` too. 

I need to ensure we set some environment variables automatically when calling `./build.bat IIS` too in that follow up PR.

**Review by v1v — APPROVED** [review]

**Pull request merged**